File: Kattis/CSharp/AlphabetSpam.cs
using System;

namespace AlphabetSpam
{
    class AlphabetSpam
    {
        static void Main(string[] args)
        {
            string sentence = Console.ReadLine();
            double whitespace = 0.0, lowercase = 0.0, uppercase = 0.0, symbols = 0.0;
            double length = sentence.Length;

            for (int i = 0; i < length; i++)
            {
                char character = sentence[i];
                if (character == '_')
                    whitespace++;
                else if (char.IsUpper(character))
                    uppercase++;
                else if (char.IsLower(character))
                    lowercase++;
                else
                    symbols++;
            }
            Console.WriteLine((whitespace / length) + "\n" + (lowercase / length) + "\n" + (uppercase / length) + "\n" + (symbols / length));
            Console.ReadKey();
        }
    }
}

File: Kattis/Python/AlphabetSpam.py
sentence = input()
length = len(sentence)
whitespace = 0.0
lowercase = 0.0
uppercase = 0.0
symbols = 0.0

for i in sentence:
    if i == "_":
        whitespace += 1
    elif i.islower():
        lowercase += 1
    elif i.isupper():
        uppercase += 1
    else:
        symbols += 1

print("%0.16f" % (whitespace / length))
print("%0.16f" % (lowercase / length))
print("%0.16f" % (uppercase / length))
print("%0.16f" % (symbols / length))
